From a63382c6164e5c856057a394f53c91cb0d79222f Mon Sep 17 00:00:00 2001
From: Sam Shih <sam.shih@mediatek.com>
Date: Fri, 2 Jun 2023 13:06:05 +0800
Subject: [PATCH] 
 [slow-speed-io][999-2102-trng-Add-trng-support-for-mt7988.patch]

---
 drivers/char/hw_random/mtk-rng.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/char/hw_random/mtk-rng.c b/drivers/char/hw_random/mtk-rng.c
index 75fca4cef..878170c15 100644
--- a/drivers/char/hw_random/mtk-rng.c
+++ b/drivers/char/hw_random/mtk-rng.c
@@ -231,6 +231,10 @@ static const struct mtk_rng_of_data mt7986_rng_data = {
 	.rng_version = 1,
 };
 
+static const struct mtk_rng_of_data mt7988_rng_data = {
+	.rng_version = 2,
+};
+
 static const struct mtk_rng_of_data mt7623_rng_data = {
 	.rng_version = 1,
 };
@@ -238,6 +242,7 @@ static const struct mtk_rng_of_data mt7623_rng_data = {
 static const struct of_device_id mtk_rng_match[] = {
 	{ .compatible = "mediatek,mt7981-rng", .data = &mt7981_rng_data },
 	{ .compatible = "mediatek,mt7986-rng", .data = &mt7986_rng_data },
+	{ .compatible = "mediatek,mt7988-rng", .data = &mt7988_rng_data },
 	{ .compatible = "mediatek,mt7623-rng", .data = &mt7623_rng_data },
 	{},
 };
-- 
2.34.1

